network/jbuild

```diff
@@ -32,7 +32,7 @@ let () = Printf.ksprintf Jbuild_plugin.V1.send {|
  ((name xcp_network_interface)
   (public_name xcp.network.interface)
   (modules (network_interface))
-  (flags (:standard -w -39 %s))
+  (flags (:standard -w -39 -w -34 -w -27 %s))
```

Comment:
Please do not silence warnings 27 and 34. In general they can easily be fixed prefixing variables or types with `_` or deleting that if they are unused or unexposed. They can also be silenced locally if really needed.

Reply:
`-w -34` has been deleted, but I cannot delete `-w -27` because if delete it the compiler will complains 
```
File "network/network_interface.ml", line 116, characters 0-611:
Warning 27: unused variable x.
```
Any idea about the warning? I think it comes from ppx_core

Reply:
The warning above should not break the compilatio though, we could live with the warning. I'll have a look at the generated file

Reply:
That is a bug of `ocaml-rpc` that does generate some code like:
```ocaml
typ_of_interface_config_t =
  let open! Rpc.Types in
      Struct
        ({
           fields =
             [BoxedField interface_config_t_ipv4_conf;
             BoxedField interface_config_t_ipv4_gateway;
             BoxedField interface_config_t_ipv6_conf;
             BoxedField interface_config_t_ipv6_gateway;
             BoxedField interface_config_t_ipv4_routes;
             BoxedField interface_config_t_dns;
             BoxedField interface_config_t_mtu;
             BoxedField interface_config_t_ethtool_settings;
             BoxedField interface_config_t_ethtool_offload;
             BoxedField interface_config_t_persistent_i];
           sname = "interface_config_t";
           version = None;
           constructor =
             (fun getter  ->
                let open Rresult.R in
                  (match getter.Rpc.Types.fget "persistent_i"
                           (let open! Rpc.Types in Basic Bool)
                   with
                   | Result.Ok x as y -> y
                   | Result.Error _ -> Result.Ok false) >>=
                    (fun interface_config_t_persistent_i  ->
                       (match getter.Rpc.Types.fget "ethtool_offload"
                                (let open! Rpc.Types in
                                   Dict
                                     (String,
                                       (let open! Rpc.Types in Basic String)))
                        with
                        | Result.Ok x as y -> y
                        | Result.Error _ -> Result.Ok [("lro", "off")]) >>=
                         (fun interface_config_t_ethtool_offload  ->
                            (match getter.Rpc.Types.fget "ethtool_settings"
                                     (let open! Rpc.Types in
                                        Dict
                                          (String,
                                            (let open! Rpc.Types in
                                               Basic String)))
                             with
                             | Result.Ok x as y -> y
                             | Result.Error _ -> Result.Ok []) >>=
                              (fun interface_config_t_ethtool_settings  ->
                                 (match getter.Rpc.Types.fget "mtu"
                                          (let open! Rpc.Types in Basic Int)
[...]
```

We'll fix it there, please leave the warning on and instead modify the `Makefile` as follows:

Change
```
build:
            jbuilder build @install --dev -j $$(getconf _NPROCESSORS_ONLN)
release:
            jbuilder build @install
```

into 
```
release:
            jbuilder build @install -j $$(getconf _NPROCESSORS_ONLN)

build:
            jbuilder build @install --dev
```

Reply:
I have made a PR to fix the issue upstream: https://github.com/mirage/ocaml-rpc/pull/77
We will get it in xenserver after the `opam-next` merge

Reply:
Thanks @mseri ! BTW, can you show me how to get the code that generated by ocaml-rpc that as you showed above so I can check them when I get a compilation error?

Reply:
After `jbuilder build`, if you enter into `_build/default`, you can run the ppx rewriters generated by jbuilder as follows
```
$ .ppx/ppx_deriving_rpc/ppx.exe --impl network/network_interface.ml > network_interface.generated.ml
```

Depending on the file, you may need to use `ppx_deriving_rpc+ppx_sexp_conv` instead of `ppx_deriving_rpc`, or some other combination or rewriters, but you get the idea

Reply:
@mseri However I cannot find `.ppx/ppx_deriving_rpc/ppx.exe` after executing `jbuilder build` in `xcp-idl` repo.